pkg/registry/registry.go

```diff
@@ -368,29 +369,24 @@ func (a *appsList) fetch(r *registryFetchState, fetchAll bool) error {
 	return nil
 }
 
+func ParseSlug(raw json.RawMessage) string {
+	var slug string
+	if err := json.Unmarshal(raw, &slug); err != nil {
+		return ""
+	}
+	return slug
+}
+
 func (a *appsList) Paginated(sortBy string, reverse bool, limit int) *AppsPaginated {
-	sortBySlug := sortBy == "slug"
 	sort.Slice(a.list, func(i, j int) bool {
-		vi := a.list[i]
-		vj := a.list[j]
-		var less, equal bool
-		switch valA := vi[sortBy].(type) {
-		case string:
-			valB := vj[sortBy].(string)
-			less = valA < valB
-			if !sortBySlug && !less {
-				equal = valA == valB
-			}
-		case int:
-			valB := vj[sortBy].(int)
-			less = valA < valB
-			if !sortBySlug && !less {
-				equal = valA == valB
-			}
-		}
+		valA := a.list[i][sortBy]
+		valB := a.list[j][sortBy]
+		cmp := bytes.Compare([]byte(valA), []byte(valB))
+		equal := cmp == 0
```

Comment:
This could be true even even in the case where `sortBy` is `"slug"` when it would have been false before.
I'm still having a hard time figuring out if that's an issue or not as I'm not sure I understand what this lambda is meant to do.